### PR TITLE
Makman2/nextgen3-3

### PR DESCRIPTION
**Pull request by Makman2**

### CODE REVIEW

File: tests/core/GraphsTest.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    except KeyError:
+        return set()
+
+    if isinstance(item, Iterable):
+        return item
+    else:
+        return {item}
+
+
+def traverse_graph_on_cyclic_graph_test(graph, start_nodes):
```
Comment (meetmangukiya) [inline tests/core/GraphsTest.py]: start function name with test?

File: tests/core/GraphsTest.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    """
+
+    def test_function(self):
+        with self.assertRaises(CircularDependencyError) as cm:
+            traverse_graph(start_nodes, partial(get_successive_nodes, graph))
+
+    return test_function
+
+
+def traverse_graph_test(graph, start_nodes, expected):
```
Comment (meetmangukiya) [inline tests/core/GraphsTest.py]: same, start function name with start?

File: coalib/core/Graphs.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            path.add(node)
+
+            for subnode in get_successive_nodes(node):
+                run_on_edge(node, subnode)
+
+                if subnode in path:
+                    raise CircularDependencyError(subnode)
+
+                visit(subnode)
+
```
Comment (meetmangukiya) [inline coalib/core/Graphs.py]: to much whitespace(IMO) ?

**Comment by meetmangukiya** [comment]
I left some comments, but you may ignore them all, nothing significant :P

File: tests/core/GraphsTest.py (hunk 1 above)
Comment (Makman2) [inline tests/core/GraphsTest.py]: that's a function that generates tests, so I thought this convention is more appropriate^^

File: tests/core/GraphsTest.py (hunk 2 above)
Comment (Makman2) [inline tests/core/GraphsTest.py]: see other comment

File: coalib/core/Graphs.py (hunk 3 above)
Comment (Makman2) [inline coalib/core/Graphs.py]: you think so? Imo it's okay :)

**Comment by Udayan12167** [comment]
ack https://github.com/coala/coala/pull/3953/commits/3f88f662098cd51d55bdfa68814e744d44ecda3d

**Comment by Udayan12167** [comment]
ack 1f1594c

**Comment by Udayan12167** [comment]
@Makman2 Can you rebase?

**Comment by Makman2** [comment]
reack 359d9e0 2857189

**Comment by Makman2** [comment]
@rultor merge

**Pull request merged**